### PR TITLE
Warning when tip is more or equal to 50% of the contribution

### PR DESCRIPTION
**Pull request by znarf**
Proposal for https://github.com/opencollective/opencollective/issues/4153

### CODE REVIEW

File: components/contribution-flow/index.js [hunk 1]
```diff
@@ -571,10 +571,10 @@ class ContributionFlow extends React.Component {
             return false;
           } else if (!isNil(tier?.availableQuantity) && stepDetails.quantity > tier.availableQuantity) {
             return false;
-          } else if (stepDetails.platformContribution && stepDetails.platformContribution > stepDetails.amount) {
+          } else if (stepDetails.platformContribution && stepDetails.platformContribution / stepDetails.amount >= 0.5) {
```
Comment (Betree) [inline components/contribution-flow/index.js:574]: From the `noPaymentRequired` variable above, I assume that `stepDetails.amount` can be `=== 0` here (=> division by 0). I don't think tips are enabled for free tickets, but to be on the safe side I would change that to something like:

```suggestion
          } else if (stepDetails.platformContribution && (!stepDetails.amount || stepDetails.platformContribution / stepDetails.amount >= 0.5)) {
```

Comment (znarf) [reply components/contribution-flow/index.js:574]: Can I instead only do the check if `stepDetails.amount` is truthy?

Comment (Betree) [reply components/contribution-flow/index.js]: Yes, that's what the suggestion above does:

```es6
!stepDetails.amount || stepDetails.platformContribution / stepDetails.amount >= 0.5
```

Comment (znarf) [reply components/contribution-flow/index.js]: Ok, I wrote it differently.

**Pull request merged**